### PR TITLE
[MM-28124] api4/post: return error at getFlaggedPostsForUser

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
If we get an error from the methods `(*App).GetFlaggedPosts` we need to stop execution and return so that we don't get nil pointer dereference. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28124

### CODE REVIEW

**Review by agnivade — CHANGES_REQUESTED** [review]
Would it also be possible to add a test for this? :)

**Comment by agnivade** [comment]
Hmm .. the test is gone now :thinking:

File: api4/post_test.go [hunk 1]
```diff
@@ -1354,6 +1357,26 @@ func TestGetFlaggedPostsForUser(t *testing.T) {
 
 	_, resp = th.SystemAdminClient.GetFlaggedPostsForUser(user.Id, 0, 10)
 	CheckNoError(t, resp)
+
+	mockStore := mocks.Store{}
+	mockPostStore := mocks.PostStore{}
+	mockPostStore.On("GetFlaggedPosts", mock.Anything, mock.Anything, mock.Anything).Return(nil, errors.New("some-error"))
```
Comment (agnivade) [inline api4/post_test.go:1363]: Nit: The params are builtin types - string, int, int. We could use `mock.AnythingOfType("string"), mock.AnythingOfType("int"), mock.AnythingOfType("int")`

File: api4/post_test.go [hunk 2]
```diff
@@ -1354,6 +1357,26 @@ func TestGetFlaggedPostsForUser(t *testing.T) {
...
+	mockPostStore.On("ClearCaches", mock.Anything).Return()
```
Comment (agnivade) [inline api4/post_test.go:1364]: `ClearCaches` doesn't take any params. Is there a reason we need to use mock.Anything?

**Review by agnivade — APPROVED** [review]
Nothing blocking.

File: api4/post_test.go [hunk 3]
```diff
@@ -1354,6 +1357,26 @@ func TestGetFlaggedPostsForUser(t *testing.T) {
...
+	mockStore.On("User").Return(th.App.Srv().Store.User())
+	mockStore.On("Scheme").Return(th.App.Srv().Store.Scheme())
+	mockStore.On("Post").Return(&mockPostStore)
+	mockStore.On("FileInfo").Return(th.App.Srv().Store.FileInfo())
+	mockStore.On("Webhook").Return(th.App.Srv().Store.Webhook())
+	mockStore.On("System").Return(th.App.Srv().Store.System())
+	mockStore.On("License").Return(th.App.Srv().Store.License())
+	mockStore.On("Role").Return(th.App.Srv().Store.Role())
+	mockStore.On("Close").Return(nil)
+	th.App.Srv().Store = &mockStore
```
Comment (mgdelacroix) [inline api4/post_test.go:1376]: nit: can we restore the `Store` value after this test case just in case someone decides to add another one after?

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1:

**Comment by isacikgoz** [comment]
@lindalumitchell This is a fix where an error handling code is written in wrong order. Automated tests should be sufficient for this one.

**Review by lindalumitchell — APPROVED** [review]

**Pull request merged**